third_party/java/arcs/build_defs/internal/manifest.bzl

```diff
@@ -73,6 +73,44 @@ def arcs_manifest_proto(name, src, deps = [], out = None, visibility = None):
         sigh_cmd = "manifest2proto --outdir $(dirname {OUT}) --outfile $(basename {OUT}) {SRC}",
     )
 
+def arcs_proto_plan(name, src, recipe = None, deps = []):
```

Comment:
The `recipe2plan` rule allows the user to accept multiple manifests as `srcs`. Should we follow that convention? This can be done by calling sigh in a loop and outputing a list of proto artifacts (in recipie2plan, these are bundled in a .jar. Is there a bazel rule to merge protos?). 

With the `--recipe` flag added to the other PR, it seems like you'd prefer to be more granular - one proto per *recipe* over one per *manifest* (over one per *manifest closure*).

Reply:
Later, when re-writing recipe2plan, we could invoke `arcs_proto_plan` once for each manifest `src`, then code-generate plans one-per-proto. However, I think this would likely slow the system down due to #4604.

Reply:
The case I am after here is one `.arcs` file only. There is a file with a query (expressed in a recipe) and I want to get a proto for that query. The rest of the manifests are given in deps.

We can refactor if we get to kotlin codegen in Kotlin - but then you will see what exactly you need and you will be able to restructure it exactly to your needs.